### DISCUSSION TITLE
Support for different messages in commit messages and PR titles.

### DISCUSSION THREAD

**Discussion by SPGoding** [renovatebot/renovate#24726] [Suggest an Idea]
# Why I Request This Feature

I want to use [Gitmoji](https://gitmoji.carloscuesta.me/) in my commit messages and [here is my config](https://github.com/SPGoding/renovate-config-gitmoji/blob/master/package.json).

The commit message is perfect:

![](https://i.loli.net/2019/06/15/5d04ebe9b33c294789.png)

But the PR titles are really ugly because `:arrow_up:` is not parsed (Maybe this is a bug of GitHub):

![](https://i.loli.net/2019/06/15/5d04ec11dfaac22306.png)

To prevent this, I expect Renovate to provide methods of setting different messages in git commits and PR titles, like `commitMessagePrefix` for commit messages and `prTitlePrefix` for PR titles.

# Possible Changes

New config options:

| key | type | default | description |
| - | - | - | - |
| `prTitle` | string | `"{{{prTitlePrefix}}} {{{prTitleAction}}} {{{prTitleTopic}}} {{{prTitleExtra}}} {{{prTitleSuffix}}}"` | Message to use for pull request titles. |
| `prTitleAction` | string | `"{{{commitMessageAction}}}"` | Action verb to use in PR titles. |
| `prTitleExtra` | string | `"{{{commitMessageExtra}}}"` | Extra description used after the pr title topic - typically the version |
| `prTitlePrefix` | string | `"{{{commitMessagePrefix}}}"` | Prefix to add to start of PR titles. |
| `prTitleSuffix` | string | `"{{{commitMessageSuffix}}}"` | Suffix to add to end of PR titles. |
| `prTitleTopic` | string | `"{{{commitMessageTopic}}}"` | The upgrade topic/noun used in PR titles |

So it's possible to write something like this in the renovate config file:

```
{
    "commitMessagePrefix": ":arrow_up:",
    "prTitlePrefix": "⬆️",
    ...
}
```

When all `prTitleXXX`s are omitted, the PR title will be exactly the same as the commit message (current behaviour).

:heart: Thanks!

**Comment by rarkins** [comment]
So you want emoji for commit messages but Unicode for PR titles?

**Comment by SPGoding** [comment]
> So you want emoji for commit messages but Unicode for PR titles?

Yes!

**Comment by rarkins** [comment]
This isn't possible via config alone - will need a new feature - so I'm moving this issue to the main repo
